doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst

```diff
@@ -179,7 +179,23 @@ Following are the various LwM2M carrier library events:
 
-    * :c:macro:`LWM2M_CARRIER_ERROR_BOOTSTRAP` - This error is generated from the :c:func:`modem_key_mgmt_write` function, if the :c:member:`value` field is negative. If the :c:member:`value` field is 0, it indicates that the bootstrap sequence has failed. If this error persists, contact your carrier.
+    * :c:macro:`LWM2M_CARRIER_ERROR_BOOTSTRAP` - This error is generated during the bootstrap procedure.
+
+      +--------------------------------------------------------+-----------------------------------------------------------------------------+--------------------------------------------------+
+      | Errors                                                 | More information                                                            | Recovery                                         |
+      |                                                        |                                                                             |                                                  |
+      +========================================================+=============================================================================+==================================================+
+      | Retry limit for connecting to the bootstrap            | Common reason for this failure can be incorrect URI or PSK,                 | Library will retry after next device reboot.     |
+      | server has been reached (``-ETIMEDOUT``).              | or the server is unavailable (for example temporary network issues).        |                                                  |
+      |                                                        | If this error persists, contact your carrier.                               |                                                  |
+      +--------------------------------------------------------+-----------------------------------------------------------------------------+--------------------------------------------------+
+      | Failure to provision the PSK                           | If the link is forced up by the application during the bootstrap procedure  | Library will retry after 24 hours.               |
+      | needed for the bootstrap procedure.                    | the error will be ``-EACCES``. Verify that the LwM2M library is controlling |                                                  |
+      |                                                        | the link until the :c:macro:`LWM2M_CARRIER_EVENT_LTE_READY` event is sent.  |                                                  |
+      +--------------------------------------------------------+-----------------------------------------------------------------------------+--------------------------------------------------+
+      | Failure to fetch MSISDN or ICCID values (``-EFAULT``). | These values are received from the network.                                 | Library will retry upon next network connection. |
```

Comment:
`MSISDN` will be received from the network for some carriers.
`ICCID` is fetched from SIM.